File: app/src/main/java/com/example/getfood/ui/foodmenu/FoodMenuDisplayActivityPresenter.java
package com.example.getfood.ui.foodmenu;

import com.example.getfood.ui.base.BasePresenter;
import com.fazemeright.firebase_api__library.listeners.OnTaskCompleteListener;

public class FoodMenuDisplayActivityPresenter<V extends FoodMenuDisplayActivityMvpView> extends BasePresenter<V> implements FoodMenuDisplayActivityMvpPresenter<V> {

    public FoodMenuDisplayActivityPresenter() {
    }

    @Override
    public String getUserEmail() {
        return apiManager.getCurrentUserEmail();
    }

    @Override
    public void sendPasswordResetEmail() {
        apiManager.sendPasswordResetEmail(getUserEmail(), new OnTaskCompleteListener() {
            @Override
            public void onTaskSuccessful() {
                signOutUser();
                getMvpView().onPasswordResetEmailSentSuccessfully();
            }

            @Override
            public void onTaskCompleteButFailed(String errMsg) {
                getMvpView().onPasswordResetEmailSentFailed(errMsg);
            }

            @Override
            public void onTaskFailed(Exception e) {

            }
        });
    }

    @Override
    public void openCart() {
        if (dataManager.getCartSize() > 0) {
            getMvpView().openCart();
        } else {
            getMvpView().cantOpenCart();
        }
    }

    @Override
    public void signOutUser() {
        apiManager.forceSignOutUser();
    }
}

File: app/src/main/java/com/example/getfood/ui/foodmenu/FoodMenuDisplayActivityPresenter.kt
package com.example.getfood.ui.foodmenu

import com.example.firebase_api_library.listeners.OnTaskCompleteListener
import com.example.getfood.ui.base.BasePresenter

class FoodMenuDisplayActivityPresenter<V : FoodMenuDisplayActivityMvpView?> : BasePresenter<V>(), FoodMenuDisplayActivityMvpPresenter<V> {
    override fun sendPasswordResetEmail() {
        apiManager!!.sendPasswordResetEmail(userEmail, object : OnTaskCompleteListener {
            override fun onTaskSuccessful() {
                signOutUser()
                mvpView?.onPasswordResetEmailSentSuccessfully()
            }

            override fun onTaskCompleteButFailed(errMsg: String?) {
                mvpView?.onPasswordResetEmailSentFailed(errMsg)
            }

            override fun onTaskFailed(e: Exception?) {}
        })
    }

    override fun openCart() {
        if (dataManager.cartSize > 0) {
            mvpView?.openCart()
        } else {
            mvpView?.cantOpenCart()
        }
    }

    override fun signOutUser() {
        apiManager!!.forceSignOutUser()
    }

    override val userEmail: String?
        get() = apiManager?.userEmail
}